tests/catch/unit/device/hipDeviceSetGetCacheConfig.cc

```diff
@@ -1,45 +1,112 @@
-OUT OF OR INN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 THE SOFTWARE.
 */
+
+#include <array>
+
 #include <hip_test_common.hh>
-/**
- * hipDeviceGetCacheConfig tests
- * Scenario1: Validates if pConfig = nullptr returns hip error code.
- * Scenario2: Validates if the value returned by hipDeviceGetCacheConfig is valid.
- */
-TEST_CASE("Unit_hipDeviceGetCacheConfig_NegTst") {
-  // Scenario1
-  REQUIRE_FALSE(hipSuccess == hipDeviceGetCacheConfig(nullptr));
+#include <threaded_zig_zag_test.hh>
+
+namespace {
+constexpr std::array<hipFuncCache_t, 4> kCacheConfigs{
+    hipFuncCachePreferNone, hipFuncCachePreferShared, hipFuncCachePreferL1,
+    hipFuncCachePreferEqual};
+}  // anonymous namespace
+
+
+TEST_CASE("Unit_hipDeviceSetCacheConfig_Positive_Basic") {
+  const auto device = GENERATE(range(0, HipTest::getDeviceCount()));
+  HIP_CHECK(hipSetDevice(device));
+  INFO("Current device is: " << device);
+
+  const auto cache_config =
+      GENERATE(from_range(std::begin(kCacheConfigs), std::end(kCacheConfigs)));
+#if HT_AMD
+  HIP_CHECK_ERROR(hipDeviceSetCacheConfig(cache_config), hipErrorNotSupported);
+#elif HT_NVIDIA
+  HIP_CHECK(hipDeviceSetCacheConfig(cache_config));
+#endif
 }
 
-TEST_CASE("Unit_hipDeviceGetCacheConfig_FuncTst") {
-  hipFuncCache_t cacheConfig;
-  // Scenario2
-  HIP_CHECK(hipDeviceGetCacheConfig(&cacheConfig));
-  REQUIRE_FALSE(((cacheConfig != hipFuncCachePreferNone) &&
-                (cacheConfig != hipFuncCachePreferShared) &&
-                (cacheConfig != hipFuncCachePreferL1) &&
-                (cacheConfig != hipFuncCachePreferEqual)));
-  // This code exists to test the dummy implementation of
-  // hipDeviceSetCacheConfig.
-#if HT_NVIDIA
-  HIP_CHECK(hipDeviceSetCacheConfig(cacheConfig));
-#else
-  REQUIRE_FALSE(hipSuccess == hipDeviceSetCacheConfig(cacheConfig));
+TEST_CASE("Unit_hipDeviceSetCacheConfig_Negative_Parameters") {
+#if HT_AMD
+  HIP_CHECK_ERROR(hipDeviceSetCacheConfig(static_cast<hipFuncCache_t>(-1)), hipErrorNotSupported);
+#elif HT_NVIDIA
+  HIP_CHECK_ERROR(hipDeviceSetCacheConfig(static_cast<hipFuncCache_t>(-1)), hipErrorInvalidValue);
 #endif
 }
+
+TEST_CASE("Unit_hipDeviceGetCacheConfig_Positive_Default") {
+  const auto device = GENERATE(range(0, HipTest::getDeviceCount()));
+  HIP_CHECK(hipSetDevice(device));
+  INFO("Current device is: " << device);
+
+  hipFuncCache_t cache_config;
+  HIP_CHECK(hipDeviceGetCacheConfig(&cache_config));
+  REQUIRE(cache_config == hipFuncCachePreferNone);
+}
+
+#if HT_NVIDIA
```

Comment:
Please remove #if HT_NVIDIA.
Instead add Unit_hipDeviceGetCacheConfig_Positive_Basic, Unit_hipDeviceGetCacheConfig_Positive_Threaded in array DisabledTests of :

- tests\catch\hipTestMain\config\config_amd_linux_common.json
- tests\catch\hipTestMain\config\config_amd_windows_common.json

This will exclude running the mentioned tests on AMD platform for Linux, Windows & helps us to manage the disabled tests from one place.

Reply:
Done. You can proceed with the review.